library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/MainActivity.kt

```diff
@@ -39,6 +50,15 @@ internal class MainActivity :
     private val applicationName: CharSequence
         get() = applicationInfo.loadLabel(packageManager)
 
+    private val permissionRequest = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isPermissionGranted: Boolean ->
+        if (!isPermissionGranted) {
+            showToast(applicationContext.getString(R.string.chucker_notifications_permission_not_granted))
```

Comment:
I would make this a SnackBar as well with Long Duration.

Reply:
We show toasts for all other fault/errors states. The reason I used Snackbar for redirection to Settings is to have an action button, so it is convenient to just click on it and end up in settings. Here we don't have any action and to be consistent with other failed operations (like exporting stuff, etc.) I use Toast.

Reply:
Mmm I understand.

For me it was a bit odd as, when playing with the app I saw both the Toast and the SnackBar with the same error message, so I thought it was a bug.

I'm fine having it as a toast, but I would still have Duration Long as I barely had the time to read it.

Reply:
Made it long.